src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnectionFactory.cs

```diff
@@ -29,7 +28,8 @@ internal class ServiceConnectionFactory : IServiceConnectionFactory
             ConnectionDelegate connectionDelegate,
             IClientConnectionFactory clientConnectionFactory,
             IServerNameProvider nameProvider,
-            IServiceEventHandler serviceEventHandler)
+            IServiceEventHandler serviceEventHandler,
+            ServiceOptions options)
```

Comment:
Change it to IOptions<ServiceOptions> so that no need to add a WeakConnectionFactory?